Nodejs/Product/Analysis/Analysis/AnalysisLimits.cs

```diff
@@ -34,6 +34,17 @@ internal class AnalysisLimits {
             MaxObjectLiteralProperties = 50;
             MaxObjectKeysTypes = 5;
             MaxMergeTypes = 5;
+            NestedModulesLimit = 4;
```

Comment:
I'm not seeing the change that actually limits this in the product?  AnalysisTests\AnalysisFile will limit it when we run the tests, but won't do anything when running inside of VS.

Personally I would suggest not putting this in AnalysisLimits, and not having the analyzer actually be aware of this at all.  The AnalysisLimits class is really used for limits on things when we're analyzing.  But in this case we actually just don't want to analyze these files at all.  Which means we also want to avoid parsing them entirely too.  

Therefore I think the correct spot to put this limit in would be in NodejsFileNode.ShouldAnalyze.  If you return false from there then we won't even bother parsing the file which should give the best performance gain.